lib/fmrest/spyke/relation.rb

```diff
@@ -204,18 +204,26 @@ def query(*params)
         with_clone do |r|
           params = params.flatten.map { |p| normalize_query_params(p) }
 
-          if r.or_flag || r.query_params.empty?
+          if r.or_flag || r.query_params.empty? # OR
             r.query_params += params
             r.or_flag = nil
-          elsif params.length > r.query_params.length
-            params[0, r.query_params.length].each_with_index do |p, i|
-              r.query_params[i].merge!(p)
-            end
-
-            remainder = params.length - r.query_params.length
-            r.query_params += params[-remainder, remainder]
-          else
-            params.each_with_index { |p, i| r.query_params[i].merge!(p) }
+          else # AND
+            r.query_params = r
+              .query_params
+              .product(params)
+              .map do |old_params, new_params|
+                merge = old_params
+                  .transform_values { |value| Set[value] }
+                  .merge(
+                    new_params.transform_values { |value| Set[value] }
```

Comment:
should probably be `Set[value.to_s]` b/c you might be merging `"foo"` with `:foo` and `3` with `"3"`

Reply:
except that `foo: nil` queries for empty fields and `foo: ""` yields an error, so they shouldn't be merged. @pilaf do you think perhaps `foo: ""` should be equivalent to `foo: nil` (i.e, fmrest should convert both to `"="`)?